front/src/applications/stdcm/hooks/useStdcm.ts

```diff
@@ -87,75 +117,183 @@ const useStdcm = ({
-        } else {
-          setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.rejected);
-          triggerShowFailureNotification({
-            name: t('stdcm:stdcmErrors.requestFailed'),
-            message: t('translation:common.error'),
-          });
-        }
-      } catch (e) {
-        if ((e as Error).name !== 'AbortError') {
-          setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.rejected);
-          triggerShowFailureNotification(
-            castErrorToFailure(e, {
-              name: t('stdcm:stdcmErrors.requestFailed'),
-              message: t('translation:common.error'),
-            })
-          );
         }
       }
+
+      setStdcmResponses([
+        {
+          ...response,
+          rollingStock: stdcmRollingStock,
+          creationDate: new Date(),
+          speedLimitByTag: osrdconf.speedLimitByTag,
+          simulationPathSteps: osrdconf.stdcmPathSteps,
+          path: response.pathfinding_result,
+          alternativePath: false,
+        } as StdcmConflictsResponse,
+        {
+          ...resDown,
+          rollingStock: stdcmRollingStock,
+          creationDate: new Date(),
+          speedLimitByTag: osrdconf.speedLimitByTag,
+          simulationPathSteps: osrdconf.stdcmPathSteps,
+          alternativePath: true,
+        } as StdcmResponse,
+        {
+          ...resUp,
+          rollingStock: stdcmRollingStock,
+          creationDate: new Date(),
+          speedLimitByTag: osrdconf.speedLimitByTag,
+          simulationPathSteps: osrdconf.stdcmPathSteps,
+          alternativePath: true,
+          upstreamPath: true,
+        } as StdcmResponse,
+      ]);
+
+      setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.success);
+    } catch (e) {
+      setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.rejected);
+      triggerShowFailureNotification(
+        castErrorToFailure(e, {
+          name: t('stdcm:stdcmErrors.requestFailed'),
+          message: t('translation:common.error'),
+        })
+      );
+    }
+  };
+
+  // Handles any rejection or error scenarios.
+  const handleRejection = (error: unknown) => {
+    setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.rejected);
+    triggerShowFailureNotification(
+      castErrorToFailure(error, {
+        name: t('stdcm:stdcmErrors.requestFailed'),
+        message: t('translation:common.error'),
+      })
+    );
+  };
+
+  const launchStdcmRequest = async () => {
+    resetStdcmState();
+
+    const validConfig = checkStdcmConf(dispatch, t, osrdconf);
+    if (!validConfig) return;
+
+    setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.pending);
+    const payload = formatStdcmPayload(validConfig);
+
+    try {
+      const promise = postTimetableByIdStdcm(payload);
+      requestPromise.current = promise;
+
+      const response = await promise.unwrap();
+
+      if (
+        response.status === 'success' &&
+        response.simulation.status === 'success' &&
+        stdcmRollingStock
+      ) {
```

Comment:
This check is enough. I have opened an [issue](https://github.com/OpenRailAssociation/osrd/issues/8715) for editoasters (the type is not correct)
```suggestion
      if (response.status === 'success') {
```

Reply:
Done :)